apps/backend/src/modules/routes/routes.service.ts

```diff
@@ -0,0 +1,50 @@
+import { HTTPCode } from "~/libs/enums/enums.js";
+import {
+	type GetMapBoxRouteResponseDto,
+	MapBoxAPIGeometricsType,
+	MapBoxAPIProfiles,
+	type MapBoxDirectionsApi,
+} from "~/libs/modules/map-box/map-box.js";
+
+import { type PointsOfInterestService } from "../points-of-interest/points-of-interest.service.js";
+import { RoutesExceptionMessage } from "./libs/enums/routes-exception-message.enum.js";
+import { RoutesError } from "./libs/exceptions/exceptions.js";
+
+class RoutesService {
+	private mapBoxDirectionApi: MapBoxDirectionsApi;
+	private pointsOfInterestService: PointsOfInterestService;
+
+	public constructor(
+		mapBoxDirectionsApi: MapBoxDirectionsApi,
+		pointOfInterestService: PointsOfInterestService,
+	) {
+		this.pointsOfInterestService = pointOfInterestService;
+		this.mapBoxDirectionApi = mapBoxDirectionsApi;
+	}
+
+	public async buildRoute(
```

Comment:
We have a rule somewhere in the quality criteria to avoid name repetition in methods for entities, because their name is already obvious from the class name. So it should be either construct, or constructPath/contructDirections
```suggestion
	public async construct(
```